RULES/cc/cc.go

```diff
@@ -214,23 +214,43 @@ func (lib Library) arRule() core.BuildRule {
 	// There is no option to ar to always force creation of a new archive; the "c"
 	// modifier simply suppresses a warning if the archive doesn't already
 	// exist. So instead we delete the target (out) if it already exists.
-	return core.BuildRule {
-		Name: toolchain.Name() + "-ar",
-		Variables: map[string]string{
-			"command": fmt.Sprintf("rm -f $out 2> /dev/null; %s rcs $out $in", ninjaEscape(toolchain.Archiver())),
-			"description": fmt.Sprintf("AR (toolchain: %s) $out", toolchain.Name()),
-		},
+	if strings.HasSuffix(toolchain.Name(), "efi") {
+		return core.BuildRule{
+			Name: toolchain.Name() + "-lib",
+			Variables: map[string]string{
+				"command":     fmt.Sprintf("rm -f $out 2> /dev/null; %s /out:$out $in", ninjaEscape(toolchain.Archiver())),
+				"description": fmt.Sprintf("AR (toolchain: %s) $out", toolchain.Name()),
+			},
+		}
+	} else {
+		return core.BuildRule{
+			Name: toolchain.Name() + "-ar",
+			Variables: map[string]string{
+				"command":     fmt.Sprintf("rm -f $out 2> /dev/null; %s rcs $out $in", ninjaEscape(toolchain.Archiver())),
+				"description": fmt.Sprintf("AR (toolchain: %s) $out", toolchain.Name()),
+			},
+		}
 	}
 }
 
 func (lib Library) soRule() core.BuildRule {
```

Comment:
technically the name `soRule` no longer fits when this also creates DLLs

Reply:
same for `arRule`

Reply:
It creates something strange. Intermediate results are mix of linux and windows formats